.github/actions/sync-files/action.yml

```
@@ -0,0 +1,107 @@
+      run: |
+        if [[ -n "$TEMP_CONFIG_DIR" ]] && [[ -d "$TEMP_CONFIG_DIR" ]]; then
+          rm -rf "$TEMP_CONFIG_DIR"
+        fi
+    
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v4
+      if: always()
+      with:
+        name: synced-${{ inputs.type }}
+        path: artifacts/
+        retention-days: 1
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)